### DISCUSSION TITLE
FluentValidation Unable to determine the validator type.

### DISCUSSION THREAD

**Discussion by NeoCoderMatrix86** [Megabit/Blazorise#4408] [Q&A]
Hello,

I'm currently migrating a large validation framework and want to move to Fluent Validation with build in support here for Blazorise. Therefor I wrote the following code:

```
public class TrackValidator : AbstractValidator<Track>
{
	public TrackValidator() 
	{ 
		RuleFor(x => x.Position)
			.NotEmpty();
		RuleFor(x => x.Position)
			.Custom((position, context) => 
			{ 
				if (position != null)
				{
					if (position <= 0)
					{
						context.AddFailure("Position should be greater than 0!");
					}
					if (position > 99)
					{
						context.AddFailure("Position should be less than 99!");
					}
				}
			});
		RuleFor(x => x.Begin)
			.NotEmpty();
		RuleFor(x => x.Begin)
			.GreaterThanOrEqualTo(TimeSpan.Zero);
	}
}
```

Registered everything:
```
builder.Services.AddBlazorise(options =>
{
    options.Debounce = true;
    options.DebounceInterval = 300;
})
.AddBootstrap5Providers()
.AddBlazoriseFluentValidation();

builder.Services.AddValidatorsFromAssembly(typeof(App).Assembly);
```

And want to use it:
```
<Validations Model="track" HandlerType="typeof(FluentValidationHandler)">
[...]
	<Validation>
		<NumericEdit TValue="uint?" Max="99" @bind-Value="track.Position" Immediate="false">
			<Feedback>
				<ValidationError Multiline />
			</Feedback>
		</NumericEdit>
	</Validation>	
[...]
	<Validation>
		<TextEdit Text="@track.Begin?.ToString()" TextChanged="(value) => OnTimespanTextChanged(x => track.Begin = x, value)" Immediate="false">
			<Feedback>
				<ValidationError Multiline />
			</Feedback>
		</TextEdit>
	</Validation>

[...]
</Validations>
```

But with the second field (Begin), I get not the following exception:

> System.NotImplementedException: Unable to determine the validator type.
>          at Blazorise.Validation.DetermineHandlerType()
>          at Blazorise.Validation.GetValidationHandler()
>          at Blazorise.Validation.ValidateAsync(Object newValidationValue)
>          at Blazorise.Validation.InitializeInput(IValidationInput inputComponent)
>          at Blazorise.BaseInputComponent`1.<InitializeValidation>d__10[[System.String, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
>          at Blazorise.TextEdit.SetParametersAsync(ParameterView parameters)

I see that there seems to be a problem that the validation doesn't seem to get the field it should validate. Is there some idea on how I can solve this? I need to use a TextEdit as input, since users can enter timespans that are not valid (like 63:23 which results in 01:03:23 via a custom parser).

**Comment by stsrki** [comment]
Are your validator models in the same project? You need to define all assemblies in `builder.Services.AddValidatorsFromAssembly(typeof(App).Assembly);` so that FV can resolve them.

**Reply by NeoCoderMatrix86** [reply]
Yes they are. As mentioned above, I registered the Validator as you sugested (under "Registered everything:").

**Reply by NeoCoderMatrix86** [reply]
I think it has to do with finding the right field, but I'm not deep into fluent validation. Also the exception comes from Blazorise. Is there something I can do, to investigate the problem?

**Reply by stsrki** [reply]
I see what might be the problem. You don't use two ways binding so Blazor cannot resolve the field information:

```razor
<TextEdit Text="@track.Begin?.ToString()" TextChanged="(value) => OnTimespanTextChanged(x => track.Begin = x, value)" Immediate="false">
```

If you can, try to use `@bind-*` as much as possible, like:

```razor
<TextEdit @bind-Text="@track.Begin">
```

Otherwise, you can manually provide it the field info with an `Expression` parameter:

```razor
<TextEdit Text="@track.Begin?.ToString()" TextChanged="(value) => OnTimespanTextChanged(x => track.Begin = x, value)" TextExpression="@(() => track.Begin)" Immediate="false">
```

**Reply by NeoCoderMatrix86** [reply]
That's what I also thought. But now we get the problem: `track.Begin` is a `TimeSpan?` and not a string. So both of the provided solutions doesn't work. I'm currently thinking on how to solve the problem, since I need a TextEdit (or something similar) where the user can enter what ever he wants and that will get parsed afterwards (inputs like "63:12" are valid).
